fe/fe-core/src/main/java/com/starrocks/sql/ast/FieldDef.java

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.ast;
+
+import com.starrocks.analysis.ColumnPosition;
+import com.starrocks.analysis.ParseNode;
+import com.starrocks.analysis.TypeDef;
+import com.starrocks.catalog.ArrayType;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.ScalarType;
+import com.starrocks.catalog.StructField;
+import com.starrocks.catalog.StructType;
+import com.starrocks.catalog.Type;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.sql.parser.NodePosition;
+
+import java.util.List;
+
+public class FieldDef implements ParseNode {
+
+    private final String fieldName;
+    private final List<String> nestedFieldNames;
+    private final TypeDef typeDef;
+    private final ColumnPosition fieldPos;
```

Comment:
What is the difference between fieldname and nestedfieldname. Could you add some comments for this class

Reply:
> What is the difference between fieldname and nestedfieldname. Could you add some comments for this class

Done, changed in https://github.com/StarRocks/starrocks/pull/46619/commits/aade0978ca7448e2ec0ae5089e8a4ad02c0fae7e